aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -125,6 +139,21 @@ module AptosFramework::Coin {
         merge(&mut coin_store.coin, coin);
     }
 
+    /// Destroy a zero-value coin. Calls will fail if the `value` in the passed-in `token` is non-zero
+    /// so it is impossible to "burn" any non-zero amount of `Coin` without having
+    /// a `BurnCapability` for the specific `CoinType`.
+    public fun destroy_zero<CoinType>(zero_coin: Coin<CoinType>) {
+        let Coin { value } = zero_coin;
+        assert!(value == 0, Errors::invalid_argument(EDESTRUCTION_OF_NONZERO_TOKEN))
+    }
+
+    /// Extract `amount` from the passed-in `coin`, where the original token is modified in place.
+    public fun extract<CoinType>(coin: &mut Coin<CoinType>, amount: u64): Coin<CoinType> {
```

Comment:
maybe this should be split?

Reply:
Well, `split` usually returns a couple of objects, `extract` for this case makes more sense for me.

Also, we discussed another function called split above with @zekun000